python-threatexchange/threatexchange/cli/match_cmd.py

```diff
@@ -9,18 +9,23 @@
 import logging
 import pathlib
 import typing as t
-
+import tempfile
 
 from threatexchange import common
 from threatexchange.cli.fetch_cmd import FetchCommand
 from threatexchange.cli.helpers import FlexFilesInputAction
 from threatexchange.exchanges.fetch_state import FetchedSignalMetadata
 
-from threatexchange.signal_type.index import IndexMatch, SignalTypeIndex
+from threatexchange.signal_type.index import (
+    IndexMatch,
+    SignalTypeIndex,
+    IndexMatchWithRotation,
```

Comment:
blocking: I don't think this should be generalized yet, let's just move it into this file (match_cmd.py)